plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -51,183 +104,2123 @@ def get_ilo_sessions(self):
+                    dest['snmp_alert_protocol'], dest["alert_destination"])
+            }
+
+        # Get existing SNMPv3 users from server
+        server_snmpv3_users = self.get_snmpv3_users()
+        if not server_snmpv3_users["ret"]:
+            return server_snmpv3_users
+        server_snmpv3_users = server_snmpv3_users["msg"]
+
+        if dest["security_name"] not in [x["SecurityName"] for x in server_snmpv3_users]:
+            return {
+                "ret": False,
+                "msg": "security_name '%s' does not exists, destination IP: '%s'" % (
+                    dest['security_name'], dest["alert_destination"])
+            }
+
+        return {"ret": True}
+
+    def validate_alert_destinations(self, server_alert_destinations, alert_destinations):
+        # Validating input parameters for SNMP alert destinations
+        if len(server_alert_destinations) + len(alert_destinations) > 8:
+            message = "Maximum of 8 alert destinations can be added to a server..." + \
+                      "Already server has %s Alert destinations and provided %s more Alert destinations"
+            return {
+                "ret": False,
+                "msg": message % (len(server_alert_destinations), len(alert_destinations))
+            }
+        input_list = ['alert_destination', 'snmp_alert_protocol']
+        allowed_list = ['alert_destination', 'snmp_alert_protocol', 'trap_community', 'security_name']
+        for dest in alert_destinations:
+            for input in input_list:
+                if input not in dest.keys():
+                    return {
+                        "ret": False,
+                        "msg": "Input parameter '%s' is missing to create alert destination" % input
+                    }
+            for input in dest.keys():
+                if input not in allowed_list:
+                    return {
+                        "ret": False,
+                        "msg": "Unsupported parameter '%s' is provided to create alert destination" % input
+                    }
+
+            if not dest["alert_destination"]:
+                return {
+                    "ret": False,
+                    "msg": "Invalid IP address/HostName/FQDN: %s" % dest["alert_destination"]
+                }
+
+            if not isinstance(dest["alert_destination"], str):
+                return {
+                    "ret": False,
+                    "msg": "Alert Destination should be of type 'str'"
+                }
+
+            temp = str(dest["alert_destination"]).split(".")
+            if len(temp) == 4:
+                try:
+                    ipaddress.ip_address(dest["alert_destination"])
+                except Exception as e:
+                    return {
+                        "ret": False,
+                        "msg": "Invalid IP address: %s" % dest["alert_destination"]
+                    }
```

Comment:
`ipaddress.ip_address()` already validates the input, so the `split()` is unnecessary.
```suggestion
            try:
                ipaddress.ip_address(dest["alert_destination"])
            except Exception as e:
                return {
                    "ret": False,
                    "msg": "Invalid IP address: %s" % dest["alert_destination"]
                }
```
In case you want to prevent IPv6 from being used, go instead with:
```python
                ipaddress.IPv4Address(dest["alert_destination"])
```

Reply:
and what if the value is a hostname instead of an IP?

Reply:
This is not to test for IPv6, it's to check for IPv4 and other related validations. The hostname validation is also handled.